src/cachemanager.cpp

```diff
@@ -0,0 +1,284 @@
+            // Calculate SHA256 of the actual cache file content
+            QCryptographicHash hash(CACHE_HASH_ALGORITHM);
+            
+            qint64 fileSize = cacheFile.size();
+            
+            // Use centralized buffer optimization module for consistent buffer sizing
+            qint64 bufferSize = getAdaptiveVerifyBufferSize(fileSize);
+            
+            // Allocate buffer on heap for large sizes
+            std::unique_ptr<char[]> buffer = std::make_unique<char[]>(bufferSize);
+            qint64 totalBytes = 0;
+            
+            qDebug() << "Background: Cache verification using" << bufferSize/1024 << "KB buffer for" 
+                     << fileSize/(1024*1024) << "MB file";
+            
+            // Emit initial progress
+            emit verificationProgress(0, fileSize);
+            
+            while (!cacheFile.atEnd()) {
+                qint64 bytesRead = cacheFile.read(buffer.get(), bufferSize);
+                if (bytesRead == -1) {
+                    qDebug() << "Background: Error reading cache file:" << cacheFile.errorString();
+                    break;
+                }
+                hash.addData(QByteArrayView(buffer.get(), bytesRead));
+                totalBytes += bytesRead;
+                
+                // Adaptive progress update frequency based on buffer size
+                // Ensures responsive progress regardless of buffer size
+                qint64 progressInterval = std::max(256LL * 1024, bufferSize); // At least 256KB or buffer size
+                if (totalBytes % progressInterval == 0 || cacheFile.atEnd()) {
+                    emit verificationProgress(totalBytes, fileSize);
+                }
+                
+                // Allow thread interruption during long operations
+                if (QThread::currentThread()->isInterruptionRequested()) {
+                    qDebug() << "Background: Cache verification interrupted";
+                    cacheFile.close();
+                    return;
+                }
+            }
+            
+            cacheFile.close();
+            
+            QByteArray computedHash = hash.result().toHex();
+            isValid = (computedHash == expectedHash);
+            
+            qDebug() << "Background: Cache verification complete";
+            qDebug() << "  File size:" << totalBytes << "bytes";
+            qDebug() << "  Expected hash:" << expectedHash;
+            qDebug() << "  Computed hash:" << computedHash;
+            qDebug() << "  Valid:" << isValid;
+        } else {
+            qDebug() << "Background: Cache file does not exist or cannot be opened:" << fileName;
+        }
+    }
+    
+    emit verificationComplete(isValid, fileName, expectedHash);
+}
+
+void CacheVerificationWorker::checkDiskSpace()
+{
+    qDebug() << "Background: Checking disk space for cache operations";
+    
+    // Ensure cache directory exists
+    if (!ensureCacheDirectoryExists()) {
```

Comment:
Perhaps a micro-optimisation might be to record whether the cache directory existed before this call, and then delete the newly-created cache directory if there's less than 1GB free space and the cache directory didn't previously exist? (but of course the overhead of adding that code probably isn't worth it)